src/OhmmsSoA/VectorSoaContainer.h

```diff
@@ -75,13 +75,14 @@ struct VectorSoaContainer
   }
 
   ///move constructor
-  VectorSoaContainer(VectorSoaContainer&& in) : nLocal(in.nLocal), nGhosts(in.nGhosts)
+  VectorSoaContainer(VectorSoaContainer&& in) : nLocal(in.nLocal), nGhosts(in.nGhosts), nAllocated(in.nAllocated), myData(in.myData)
   {
-    nAllocated    = in.nAllocated;
-    myData        = in.myData;
+    free(myAlloc);
```

Comment:
I feel bad using free in C++ code. Is the std::vector implemented in a same way?

Reply:
Both gnu and llvm avoid use of malloc or free in their vector source. Worth changing it.

Reply:
Probably just put in the initializer list of the move constructor `myAlloc(std::move(in.myAlloc))`

Reply:
The use of free isn't limited to here unfortunately.  I want to put this off until some other work on OhmmsVector since it requires changes there as well.